patches/llvm-project/0015-Multilib-support-for-libraries-with-exceptions.patch

```diff
@@ -0,0 +1,27 @@
+From 26fe33c4db8b786077e34966a4d9e8b40dc90745 Mon Sep 17 00:00:00 2001
+From: Piotr Przybyla <piotr.przybyla@arm.com>
+Date: Wed, 29 Nov 2023 14:05:00 +0000
+Subject: [PATCH 15/15] Multilib support for libraries with exceptions
+
+---
+ clang/lib/Driver/ToolChains/BareMetal.cpp | 4 ++++
+ 1 file changed, 4 insertions(+)
+
+diff --git a/clang/lib/Driver/ToolChains/BareMetal.cpp b/clang/lib/Driver/ToolChains/BareMetal.cpp
+index 42c8336e626c..d887c650748d 100644
+--- a/clang/lib/Driver/ToolChains/BareMetal.cpp
++++ b/clang/lib/Driver/ToolChains/BareMetal.cpp
+@@ -179,6 +179,10 @@ static void findMultilibsFromYAML(const ToolChain &TC, const Driver &D,
+   if (!MB)
+     return;
+   Multilib::flags_list Flags = TC.getMultilibFlags(Args);
++  if (TC.getRTTIMode() == ToolChain::RTTIMode::RM_Disabled) {
++    Flags.push_back("-fno-exceptions");
```

Comment:
I believe exceptions and RTTI can be enabled/disabled independently - we can choose to combine as a library variant, but we should not hardcode them here, I think.

Reply:
When exceptions are enabled without rtti one of the tests fails. This is the reason we have it always enabled/disabled at the same time. More to say we ship only two types of library. Possibility to enable/disable rtti and exceptions independently would require to ship more library variants.

Reply:
One thing is what we do and the other is what other users of the upstream code will do. Even if there are dependencies between the two elsewhere, I agree the checks should be separate here.

Could we also have explicit flags for the positive case? I mean `-fexceptions` and `-frtti`?

As the current version works for us, I think it can be fixed during upstream review.

Reply:
We will have one set of libraries for now, but this should be addressed in the multilib mapping file - the driver should allow splitting them later, i.e. should allow configuring exceptions and rtti flags independently.

Reply:
As for the test. I think the test should be fixed upstream with something like:
```
// REQUIRES: rtti
```